rust/chains/optics-ethereum/src/retrying.rs

```diff
@@ -0,0 +1,81 @@
+use std::{fmt::Debug, str::FromStr, time::Duration};
+
+use async_trait::async_trait;
+use ethers::{
+    prelude::{JsonRpcClient, ProviderError},
+    providers::Http,
+};
+use serde::{de::DeserializeOwned, Serialize};
+use thiserror::Error;
+use tokio::time::sleep;
+
+/// A naive exponential backoff provider
+#[derive(Debug, Clone)]
+pub struct RetryingProvider {
+    inner: Http,
+    max_requests: usize,
```

Comment:
there is a setter, but no getter. awkward!

```suggestion
    pub max_requests: usize,
```

will let anyone use it or change it if they need.